doc/hpc/index.rst

```diff
@@ -1,3 +1,5 @@
+.. _hpcoverview:
```

Comment:
Ben left a comment on my [PR](https://github.com/arbor-sim/arbor/pull/1227) a few weeks ago about reference naming conventions. He's been using `filename-sectionname`, and I think that is a good way to go. I've been meaning to go through our existing references and making the changes there too.
If you agree, we can rename this to `hpc-overview`

Reply:
I'll add this to a new howto-write-docs file.